source/common/http/header_utility.cc

```diff
@@ -20,6 +20,15 @@
 namespace Envoy {
 namespace Http {
 
+namespace {
+
+const Regex::CompiledGoogleReMatcher& connectUdpPathMatcher() {
```

Comment:
A regex seems pretty heavyweight for something this simple. I'd assume we'd get noticeably better performance by checking whether the path is prefixed by `/.well-known/masque/udp/`, removing the prefix and then splitting on `/`.

That said, this could be a premature optimization. Is there precedent for regex vs split in the codebase currently?

Reply:
Regex is used in Envoy mainly for rewriting headers based on the user configuration. I agree that in this case, regex could be an overkill. I think this could make more sense in a later stage if we want to support more custom URI Templates. I will remove regex as you suggested.